--- pyperl-1.0.1d/Python-Object/Makefile.PL~	2012-06-14 12:21:30.505472673 +0200
+++ pyperl-1.0.1d/Python-Object/Makefile.PL	2012-06-14 12:21:32.586446657 +0200
@@ -9,9 +9,10 @@ my @libs;
 
 push(@define, "MULTI_PERL") if -f "../MULTI_PERL";
 
+push(@libs, $pylib);
+
 if (-f "../BOOT_FROM_PERL") {
     push(@define, "BOOT_FROM_PERL");
-    push(@libs, $pylib);
     push(@extra, OBJECT => "Object.o $pysitelib/perl.so");
 }
 
